toolchain/parse/handle_match.cpp

```diff
@@ -0,0 +1,232 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/parse/context.h"
+
+namespace Carbon::Parse {
+
+namespace {
+auto HandleStatementsBlockStart(Context& context, State finish,
```

Comment:
We typically use `static` on the function rather than an anonymous namespace:

```suggestion
static auto HandleStatementsBlockStart(Context& context, State finish,
```